src/librustc_target/abi/call/sparc64.rs

```diff
@@ -20,14 +20,7 @@ fn is_homogeneous_aggregate<'a, Ty, C>(cx: &C, arg: &mut ArgType<'a, Ty>)
             RegKind::Vector => arg.layout.size.bits() == 128
         };
 
-        if valid_unit {
-            Some(Uniform {
-                unit,
-                total: arg.layout.size
-            })
-        } else {
-            None
-        }
+        valid_unit.to_option(Uniform { unit, total: arg.layout.size })
```

Comment:
I've seen the same logic (up to `RegKind::Vector =>`) 4 times now so it might be worth de-duplicating that with a function somewhere.